src/Datadog.Trace/Agent/Transports/HttpStreamRequest.cs

```diff
@@ -38,11 +40,26 @@ public void AddHeader(string name, string value)
             _headers.Add(name, value);
         }
 
+        public async Task<IApiResponse> PostAsJsonAsync(IEvent events)
+        {
+            using var memoryStream = new MemoryStream();
+            using var streamWriter = new StreamWriter(memoryStream);
+            var json = JsonConvert.SerializeObject(events);
+            streamWriter.Write(json);
+            var buffer = memoryStream.GetBuffer();
+            return await PostSegmentAsync(new ArraySegment<byte>(buffer, 0, (int)memoryStream.Length)).ConfigureAwait(false);
+        }
+
         public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces)
+        {
+            return await PostSegmentAsync(traces).ConfigureAwait(false);
+        }
```

Comment:
I know await adds a small overhead, but I think also affects error handling. With the await any exception thrown would be caught by the catch block in our AppSecAgentWriter, without it, I don't think it would. Perhaps there's another error handling mechanism already in place?

Reply:
Yeah, it does change things but I don't think it would be an impact here? The _calling_ code of this method should be awaiting it, so _not_ awaiting this won't be an issue I think 🤔

Reply:
Yes, you're right as it's pass though the async await doesn't have any benefit.  Corrected in e19f8522